### PR TITLE
Add collaborative type forum

### PR DESCRIPTION
**Pull request by franciclo**
Any user can create proposals, proposals created by users without other privileges will be automatically published.

### CODE REVIEW

File: lib/api-v2/middlewares/topics.js [hunk 1]
```diff
@@ -124,6 +125,11 @@ exports.parseUpdateableKeys = function parseUpdateableKeys (req, res, next) {
     }
   })
 
+  // Auto publish topic for common user proposal
+  if (req.forum.visibility === 'collaborative' && !forumPrivileges.canChangeTopics(req.forum, req.user)) {
```
Comment (mjlescano) [inline lib/api-v2/middlewares/topics.js]: no need to verify `forum.visibility`

File: lib/privileges/topic.js [hunk 2]
```diff
@@ -14,14 +14,19 @@ const execAll = require('lib/utils/exec-all')
...
+    if (forum.hasVisibility('collaborative') && topic.isOwner(user)) return true
+    return false
   },
 
   canDelete: function canDelete (forum, user, topic) {
     if (!config.multiForum && user && user.staff) return true
     if (forum.isOwner(user) || forum.hasRole(user, 'admin')) return true
-    return forum.hasRole(user, 'author') && topic.draft
+    if (forum.hasRole(user, 'author') && topic.draft) return true
+    return topic.isOwner(user) && forum.hasVisibility('collaborative')
```
Comment (mjlescano) [inline lib/privileges/topic.js]: here, on line 27, `topic.isOwner(user)` already returned `true`. So, this line is the same as `return false`

File: lib/site/topic-layout/topic-article/component.js [hunk 3]
```diff
@@ -83,7 +83,17 @@ class TopicArticle extends Component {
           author={topic.author}
           authorUrl={topic.authorUrl}
           tags={topic.tags}
-          mediaTitle={topic.mediaTitle} />
+          mediaTitle={topic.mediaTitle}>
+          {
+            this.props.forum.privileges.canCreateTopics &&
+            this.props.topic.privileges.canEdit &&
```
Comment (mjlescano) [inline lib/site/topic-layout/topic-article/component.js]: Here, with `this.props.topic.privileges.canEdit` is enough.

File: lib/privileges/topic.js (hunk 2 above)
Comment (franciclo) [inline lib/privileges/topic.js]: there's no topic.isOwner on L27

**Pull request merged**